ci/dpcpp.filter

```diff
@@ -1,4 +1,5 @@
 buffer
 exceptions
+math_builtin_api
```

Comment:
The reason `math_builtin_api` is disabled here is because the Python script that generates the tests has no mechanism for disabling.

`function_objects` is disabled in the CI filter file because hipSYCL cannot compile the header that contains the actual tests as it contains `marray`. Therefore, it cannot be disabled by using `DISABLED_FOR_TEST_CASE` and implementation-specific `#ifdef`s are required, which I understood to be undesirable.

Reply:
> implementation-specific `#ifdef`s are required, which I understood to be undesirable.

According to my understanding it's the opposite, @psalz added support for `DISABLED_FOR_TEST_CASE` and implementation-specific `#ifdef`s specifically to avoid using CI filters.
@psalz, am I right?

Reply:
Okay, I trust your understanding of the project better than mine `;)`. I have removed `function_objects` from hipSYCL's CI filter and added the `#ifdef` workaround to deal with `marray`.

For `math_builtin_api` it is not so easy to do. I have not made that part of the PR but I believe this to require quite a big changeset, which may be out of scope for this PR.